components/brave_rewards/core/wallet_provider/connect_external_wallet.cc

```diff
@@ -118,6 +125,42 @@ ConnectExternalWallet::GetCode(
   return query_parameters.at("code");
 }
 
+void ConnectExternalWallet::CheckLinkage() {
+  if (!engine_->IsReady()) {
+    return linkage_checker_.Reset();
+  }
+
+  if (GetWalletIf(
+          *engine_, WalletType(),
+          {mojom::WalletStatus::kConnected, mojom::WalletStatus::kLoggedOut})) {
+    RequestFor<GetWallet>(*engine_).Send(base::BindOnce(
+        &ConnectExternalWallet::CheckLinkageCallback, base::Unretained(this)));
```

Comment:
We should discuss whether we ought to use `base::Unretained(this)` for new code. Should we just use weak pointers in general?

Reply:
I would love that. I personally never use `base::Unretained` unless I'm trying to adhere to the standards of an existing module.

Reply:
I'd like to avoid introducing weak pointers everywhere. It would be awesome if we could use `base::Unretained(this)` legally (https://github.com/brave/brave-core/pull/18679 would have enabled this for us, I'll look into how we could do that without it).

Reply:
I think it's going to be difficult to make sure that `Unretained` is used correctly, even with a singleton approach. Also, automated checks are going to continually warn us that we're doing something wrong.

It's a minor inconvenience to set up a weak ptr factory, but I think it's worth it.

Reply:
OK to leave this for later, and we'll set up a plan before adding any more new code.